app/tests/data_factory.py

```diff
@@ -0,0 +1,978 @@
+        last = '{}-{}'.format(
+            cls.factory_name, int(re.match(fr"{cls.factory_name}-(\d+)", last).group(1)) + 1
+        )
+        return last
+
+    def create_sample(self, name=None, db_create=False, **kwargs):
+        '''Create a data sample
+
+        Args:
+            name: string
+                Data name, if not given it creates a '{factory_name}-n' with n being incremented
+                after each function call
+            sample: string
+                sample based on the sample named found in tests.sample_data
+            **kwargs:
+                key/value pairs used to overwrite arbitrary attribute in the sample
+
+        Returns:
+            The data sample
+        '''
+        if name:
+            data_sample = self.sample_class(name=name, **kwargs)
+        else:
+            self.last = self.get_last_name(self.last)
+            data_sample = self.sample_class(name=self.last, **kwargs)
+        if db_create:
+            data_sample.create()
+        return data_sample
+
+    def create_samples(self, samples, db_create=False, kwargs_list=True, **kwargs):
+        '''Creates several samples
+
+        Args:
+            samples: integer | [string]
+                number of DB sample to create or list of sample data name
+            db_create: bool
+                create the sample in the DB
+            sample: string
+                sample name to use for all new samples
+                (NOTE: overwrite the name in samples parameter)
+            kwargs_list: bool
+                If set to true, then kwargs with list values are distributed over the samples,
+                otherwise the kwargs are passed as is to the sample.
+            **kwargs:
+                key/value pairs used to overwrite arbitrary attribute in the sample
+
+        Returns:
+            Array with the DB samples
+        '''
+        db_samples = []
+        if isinstance(samples, int):
+            samples = map(lambda i: None, range(samples))
+        for i, sample_name in enumerate(samples):
+            sample_kwargs = {
+                key: value[i] if isinstance(value, list) and kwargs_list else value for key,
+                value in kwargs.items()
+            }
+            if sample_name:
+                sample_kwargs['sample'] = sample_name
+            sample = self.create_sample(db_create=db_create, **sample_kwargs)
+            db_samples.append(sample)
+        return db_samples
+
+
+class ProviderFactory(FactoryBase):
+    factory_name = 'provider'
+    sample_class = ProviderSample
+
+
+class LinkFactory(FactoryBase):
+    # pylint: disable=arguments-differ
+    factory_name = 'relation'
+
+    def create_sample(self, rel=None, **kwargs):
+        '''Create a data sample
+
+        Args:
+            rel: string
+                Data rel, if not given it creates a 'rel-n' with n being incremented
+                after each function call
+            sample: string
```

Comment:
`sample` is not an argument of `create_sample`, probably docstring needs an update